protothread.h

```diff
@@ -12,7 +12,13 @@
 #define pt_assert(condition) do { if (PT_DEBUG) assert(condition) ; } while (0)
 
 /* standard definitions */
+#if !defined (TRUE) && !defined (FALSE)
 typedef enum bool_e { FALSE, TRUE } bool_t ;
+#else
+#include <stdbool.h>
```

Comment:
Just wondering, if TRUE and FALSE are defined, wouldn't that mean `stdbool.h` is already included? I do see the need for the `bool_t` typedef.

Reply:
The bool_t will always get type defined the question is which will it be.  If TRUE and FALSE are not defined, the it will be the enum you already had in place.  But if TRUE and FALSE are already defined, then bool_t becomes and analog of a libc bool.

The ncurses library defines TRUE and FALSE manually (something like #define TRUE 1).  Another way around this is to demand the user include protothread.h before some other, problematic include, but I have always found those kinds of workaround not palatable.

Reply:
I'm kind-of thinking, why not just remove all the conditionals on this and:

#include <stdbool.h>
typedef bool bool_t

Reply:
That's great, I like your last suggestion, always love to minimize conditionals! Why don't we just go ahead and change all bool_t to bool, since that's what stdbool.h defines, and then change all TRUE to true and all FALSE to false (and, I forgot, but if there are any reasons the code depends on bool_t being an enum type, just get rid of those reasons). I would like this stuff to be as standard as possible -- absolutely NO reason booleans should have to be different here! Would you mind making that change?